src/vcpkg/commands.update-baseline.cpp

```diff
@@ -0,0 +1,192 @@
+                                 "example of {old_value}, {new_value} is '5507daa796359fe8d45418e694328e878ac2b82f'",
+                                 "updated registry '{url}': baseline {old_value} -> {new_value}");
+    DECLARE_AND_REGISTER_MESSAGE(UpdateBaselineAlreadyUpToDate,
+                                 (msg::url, msg::value),
+                                 "example of {value} is '5507daa796359fe8d45418e694328e878ac2b82f'",
+                                 "registry '{url}' already up to date: {value}");
+    DECLARE_AND_REGISTER_MESSAGE(UpdateBaselineNoUpdate,
+                                 (msg::url, msg::value),
+                                 "example of {value} is '5507daa796359fe8d45418e694328e878ac2b82f'",
+                                 "registry '{url}' not updated: {value}");
+}
+
+namespace vcpkg::Commands
+{
+    static constexpr StringLiteral OPTION_ADD_INITIAL_BASELINE = "add-initial-baseline";
+    static constexpr StringLiteral OPTION_DRY_RUN = "dry-run";
+
+    static constexpr CommandSwitch switches[] = {
+        {OPTION_ADD_INITIAL_BASELINE, "add a `builtin-baseline` to a vcpkg.json that doesn't already have it"},
+        {OPTION_DRY_RUN, "Print out plan without execution"},
+    };
+
+    static const CommandStructure COMMAND_STRUCTURE{
+        create_example_string("x-update-baseline"),
+        0,
+        0,
+        {switches},
+    };
+
+    static void print_update_message(StringView url,
+                                     Optional<StringView> p_old_baseline,
+                                     const Optional<std::string>& new_baseline)
+    {
+        auto old_baseline = p_old_baseline.value_or("none");
+        if (auto p = new_baseline.get())
+        {
+            if (p_old_baseline.has_value() && *p == old_baseline)
+            {
+                msg::println(msgUpdateBaselineAlreadyUpToDate, msg::url = url, msg::value = old_baseline);
+            }
+            else
+            {
+                msg::println(msgUpdateBaselineUpdatedBaseline,
+                             msg::url = url,
+                             msg::old_value = old_baseline,
+                             msg::new_value = *p);
+            }
+        }
+        else
+        {
+            msg::println(msgUpdateBaselineNoUpdate, msg::url = url, msg::value = old_baseline);
+        }
+    }
+
+    void UpdateBaselineCommand::perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths) const
+    {
+        auto options = args.parse_arguments(COMMAND_STRUCTURE);
+
+        const bool add_builtin_baseline = Util::Sets::contains(options.switches, OPTION_ADD_INITIAL_BASELINE);
+        const bool dry_run = Util::Sets::contains(options.switches, OPTION_DRY_RUN);
+
+        auto configuration = paths.get_configuration();
+        const bool has_manifest = paths.get_manifest().has_value();
+        auto manifest = has_manifest ? *paths.get_manifest().get() : ManifestAndLocation{};
```

Comment:
Would this be clearer as an Optional<ManifestAndLocation> ? (Genuinely asking if you think so, not asking to do it; I have no idea if that'd be better or not)

Reply:
No, it would not (it was before and I changed it)